crates/cubecl-opt/src/passes/composite.rs

```diff
@@ -0,0 +1,124 @@
+use std::{collections::HashMap, mem::take};
+
+use cubecl_core::ir::{
+    BinaryOperator, Item, LineInitOperator, Operation, Operator, UnaryOperator, Variable,
+};
+use stable_vec::StableVec;
+
+use crate::{AtomicCounter, Optimizer};
+
+use super::OptimizationPass;
+
+pub struct CompositeMerge;
```

Comment:
It would be cool to add documentation for each optimization pass, saying what they do, why they are useful and an example of the changes made on the IR.